deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/CheckDeployEnvStatusWorker.java

```diff
@@ -0,0 +1,127 @@
+    private UtilDAO utilDAO;
+    public static final int  METRICS_AGENT_PORT = 18126;
+
+    public CheckDeployEnvStatusWorker() {
+
+    }
+
+    void processDeploys() throws Exception {
+        List<String> envIds = environDAO.getAllEnvIds();
+        Collections.shuffle(envIds);
+        //count for different deploy states, need emit them to statsboard
+        int runningDeploys = 0;
+        int failingDeploys = 0;
+        int succeedingDeploys = 0;
+        int succeededDeploys = 0;
+        int abortedDeploys = 0;
+
+        for (String envId : envIds) {
+            EnvironBean envBean = environDAO.getById(envId);
+            int index = 1;
+            int size = 100;
+            DeployFilterBean filterBean = new DeployFilterBean();
+            filterBean.setEnvIds(Arrays.asList(envBean.getEnv_id()));
+            filterBean.setPageIndex(index);
+            filterBean.setPageSize(size);
+            int maxPages = 50; // This makes us check at most 5000 deploys
+            int tocheckPages = maxPages;
+            while (tocheckPages-- > 0) {
+                DeployQueryFilter filter = new DeployQueryFilter(filterBean);
+                DeployQueryResultBean resultBean = deployDAO.getAllDeploys(filter);
+                if (resultBean.getTotal() < 1) {
+                    LOG.warn("Could not find any previous succeeded deploy in env {}", envBean.getEnv_id());
+                }
+                for (DeployBean deploy : resultBean.getDeploys()) {
+                    if (deploy.getState() == DeployState.RUNNING) {
+                        runningDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.FAILING) {
+                        failingDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.SUCCEEDING) {
+                        succeedingDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.SUCCEEDED) {
+                        succeededDeploys += 1;
+                    }
+                    else if (deploy.getState() == DeployState.ABORTED) {
+                        abortedDeploys += 1;
+                    }
+                }
+                index += 1;
+                filterBean.setPageIndex(index);
+            } 
+        }
+        emitMetrics(DeployState.RUNNING.toString(), runningDeploys);
+        emitMetrics(DeployState.FAILING.toString(), failingDeploys);
+        emitMetrics(DeployState.SUCCEEDING.toString(), succeedingDeploys);
+        emitMetrics(DeployState.SUCCEEDED.toString(), succeededDeploys);
+        emitMetrics(DeployState.ABORTED.toString(), abortedDeploys);
+    }
+
+    private void emitMetrics(String status, int stateCount) throws Exception {
+        String sentence = "";
+        Socket clientSocket = null;
+        DataOutputStream outToServer = null;
+        try {
+            clientSocket = new Socket("127.0.0.1", METRICS_AGENT_PORT);
```

Comment:
Thanks for pointing out, same reason as previously one.
This port is set by visibility team, and it will not change, I prefer pass in as  final static int.